consensus/src/chained_bft/safety/safety_rules.rs

```diff
@@ -209,36 +193,30 @@ impl<T: Payload> SafetyRules<T> {
     /// committed block, might panic otherwise.
     /// The update function is invoked whenever a system learns about a potentially high QC.
     pub fn update(&mut self, qc: &QuorumCert) {
-        //TODO: remove
         // Preferred block rule: choose the highest 2-chain head.
-        if let Some(one_chain_head) = self.block_tree.get_block(qc.certified_block_id()) {
-            if let Some(two_chain_head) = self.block_tree.get_block(one_chain_head.parent_id()) {
-                if two_chain_head.round() >= self.state.preferred_block_round() {
-                    self.state.set_preferred_block_round(two_chain_head.round());
-                }
-            }
+        if qc.certified_parent_block_round() > self.state.preferred_block_round() {
+            self.state
+                .set_preferred_block_round(qc.certified_parent_block_round());
         }
     }
 
     /// Check if a one-chain at round r+2 causes a commit at round r and return the committed
-    /// block at round r if possible
-    pub fn commit_rule_for_certified_block(
+    /// block id at round r if possible
+    pub fn commit_rule_for_certified_block<T: Payload>(
```

Comment:
We can probably change the signature to receive just the QC.

Reply:
This can be a private function?

Reply:
Made all the changes recommended. Now need another approve. :-)